client/app/telecom/pack/xdsl/modem/service/pack-xdsl-modem-service.controller.js

```diff
@@ -0,0 +1,359 @@
+      return this.$q.reject(err);
+    });
+  }
+
+  getCallWaiting() {
+    return this.OvhApiXdsl.Modem().CallWaiting().v6().get({
+      xdslId: this.$stateParams.serviceName,
+    }).$promise.then(({ data }) => {
+      if (data === 'enabled') {
+        this.service.callWaiting = true;
+      } else {
+        this.service.callWaiting = false;
+      }
+      this.service.callWaitingTmp = this.service.callWaiting;
+      this.service.isCallWaiting = true;
+      return data;
+    }).catch(() => {
+      this.service.isCallWaiting = false;
+    });
+  }
+
+  changeCallWaiting(value) {
+    if (value) {
+      this.updateCallWaiting('enabled');
+    } else {
+      this.updateCallWaiting('disabled');
+    }
+  }
+
+  updateCallWaiting(status) {
+    return this.OvhApiXdsl.Modem().CallWaiting().v6().post({
+      xdslId: this.$stateParams.serviceName,
+    },
+    {
+      callWaiting: status,
+    }).$promise.then((data) => {
+      this.mediator.setTask('changeModemCallWaitingStatus');
+      this.service.callWaitingTmp = this.service.callWaiting;
+      this.TucToast.success(this.$translate.instant('xdsl_modem_service_call_waiting_doing'));
+      return data;
+    }).catch((err) => {
+      this.service.callWaiting = this.service.callWaitingTmp;
+      this.TucToast.error(this.$translate.instant('xdsl_modem_service_call_waiting_an_error_ocurred'));
+      return this.$q.reject(err);
+    });
+  }
+
+  getContentSharing() {
+    return this.OvhApiXdsl.Modem().ContentSharing().v6().get({
+      xdslId: this.$stateParams.serviceName,
+    }).$promise.then(({ data }) => {
+      if (data === 'enabled') {
+        this.service.contentSharing = true;
+      } else {
+        this.service.contentSharing = false;
+      }
+      this.service.contentSharingTmp = this.service.contentSharing;
+      this.service.isContentSharing = true;
+      return data;
+    }).catch(() => {
+      this.service.isContentSharing = false;
+    });
+  }
+
+  changeContentSharing(value) {
+    this.$uibModal.open({
+      animation: true,
+      templateUrl: 'app/telecom/pack/xdsl/modem/service/modal/pack-xdsl-modem-service-modal.html',
+      controller: 'XdslModemServiceModalCtrl',
+      controllerAs: '$ctrl',
+    }).result.then((result) => {
+      switch (result) {
+        case 'cancel':
+          this.service.contentSharing = this.service.contentSharingTmp;
+          break;
+        default:
+          if (value) {
+            this.updateContentSharing('enabled');
+          } else {
+            this.updateContentSharing('disabled');
+          }
+          break;
+      }
+    }).finally(() => {
+      this.getFTP();
+      this.getUPnP();
+    });
+  }
+
+  updateContentSharing(status) {
+    return this.OvhApiXdsl.Modem().ContentSharing().v6().post({
+      xdslId: this.$stateParams.serviceName,
+    },
+    {
+      contentSharing: status,
+    }).$promise.then((data) => {
```

Comment:
According to the API response class:

```suggestion
    }).$promise.then((xdslTask) => {
```